### PR TITLE
[WIP] Implement conditionals.

### PR DESCRIPTION
**Pull request by jmchilton**
```
sh run_tests.sh -api lib/galaxy_test/api/cwl/test_cwl_conformance_red_v1_2.py::CwlConformanceTestCase::test_conformance_v1_2_direct_optional_null_result
```

### CODE REVIEW

**Comment by jmchilton** [comment]
03c56063be6f5c7b03d75ff53e1b3961dde2515a is the actual conditional implementing, and I think it passes every test case that includes a when with no ``pickValues``!

I was remembering pickValues being on inputs... but I was clearly wrong. All the test cases have them on workflow outputs and that is the only place they can be according to the spec. This makes implementing them more tricky for sure - some ideas for paths forward on https://github.com/common-workflow-language/galaxy/issues/126

File: lib/galaxy/workflow/modules.py [hunk 1]
```diff
@@ -1760,6 +1761,23 @@ def from_cwl(value):
         for key, value in execution_state.inputs.items():
             step_state[key] = to_cwl(value)
 
+        if when_expression is not None:
+            from cwltool.expression import do_eval
```
Comment (mvdbeek) [inline lib/galaxy/workflow/modules.py:1785]: Let's import this at the top

Comment (jmchilton) [reply lib/galaxy/workflow/modules.py:1785]: You can tell I have developed habits stemming from frequent rebasing, imports at top result in conflicts 😆 .

**Comment by mr-c** [comment]
Hey @jmchilton , what can we do to get this merged?

**Comment by mr-c** [comment]
I guess this PR can get rebased or merged with https://github.com/galaxyproject/galaxy/pull/10712 now that it has been accepted upstream?

**Comment by mr-c** [comment]
Already merged!

**Pull request closed without merging**